pkg/ccl/importccl/read_import_mysql.go

```diff
@@ -164,30 +174,40 @@ func mysqlToCockroach(
 
 func readMysqlCreateTable(
 	input io.Reader, evalCtx *tree.EvalContext, match string,
-) (*sqlbase.TableDescriptor, error) {
+) ([]*sqlbase.TableDescriptor, error) {
 	r := bufio.NewReaderSize(input, 1024*64)
 	tokens := mysql.NewTokenizer(r)
 
+	var ret []*sqlbase.TableDescriptor
 	var found []string
 	for {
 		stmt, err := mysql.ParseNext(tokens)
 		if err == io.EOF {
 			if match != "" {
 				return nil, errors.Errorf("table %q not found in file (found tables: %s)", match, strings.Join(found, ", "))
 			}
-			return nil, errors.Errorf("no table definition found")
+			if ret == nil {
+				return nil, errors.Errorf("no table definition found")
+			}
+			return ret, nil
 		}
 		if err != nil {
 			return nil, errors.Wrap(err, "mysql parse error")
 		}
 		if i, ok := stmt.(*mysql.DDL); ok && i.Action == mysql.CreateStr {
 			name := i.NewName.Name.String()
 			if match != "" && match != name {
-				found = append(found, name)
 				continue
 			}
-			id := sqlbase.ID(int(defaultCSVTableID))
-			return mysqlTableToCockroach(evalCtx, id, name, i.TableSpec)
+			id := sqlbase.ID(int(defaultCSVTableID) + len(ret))
```

Comment:
wait i forget how the table ID stuff works. IMPORT doesn't use a key rewriter. these IDs aren't valid to be inserted because they weren't allocated from a txn. so where's the thing that generates IDs for these tables that can be inserted into our kv store?

Reply:
still in import_stmt.go, the last step before it kicks off the job: it generates ids and puts them in the descriptors.

When we switch to kicking on the job without descriptors, we'll need to do that after sampling instead and then rekey the samples.